graphql/util.go

```diff
@@ -603,6 +603,21 @@ func getResourceTypeAndIdFromSubscriptionSelectors(ctx context.Context, selector
 	return idType, id, nil
 }
 
+func savePublicKey(ctx context.Context, publicKeyInput PublicKeyInput) error {
+	if doesPublicKeyNameAlreadyExist(ctx, publicKeyInput.Name) {
```

Comment:
Arjun is adding a way to update an existing public key. Does it make sense to do that in this workflow?

Reply:
Im not sure how useful that would be. A user can pass in a new public key that they can choose to save or not. If they do save it will likely be a new entry.
 The other case would be a user selects a preexisting public key that they have previously saved. I dont think there will be a case where a user would need to update a existing public key while spawning a host. It also might be confusing to allow this since it could lead to  accidental updates

Reply:
the spawnHost mutation will be used in this modal which allows saving a new public key but not updating: https://mongodb.invisionapp.com/d/main#/console/19647524/424885178/preview
https://mongodb.invisionapp.com/d/main#/console/19647524/424885180/preview